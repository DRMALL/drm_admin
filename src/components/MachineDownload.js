

import React from 'react'
import DatePicker from 'antd/lib/date-picker'
import Button from 'antd/lib/button'
import dispatch from '../actions/dispatch'
// import downLoadExcel from '../actions/downLoadExcel'
import { machineExcle } from '../commons/apis'
import moment from 'moment'
import message from 'antd/lib/message'
import request from 'superagent'

export default props => {
  const { downloadTime, machineStatusInfo } = props.state.machine
  const startTime = (new Date(moment(downloadTime[0])._d)).getTime()
  const endTime = (new Date(moment(downloadTime[1])._d)).getTime()
  const token = localStorage.getItem('token')
  function tipsExcle(){
    if(downloadTime.length) {
        request(`${machineExcle}?startTime=${startTime}&endTime=${endTime}&number=${machineStatusInfo.number}&token=${token}`)
        .then(res => {
          const result = res.body.data
          window.location.href = result
        })
        .catch(res => console.error(res))
    }
    else message.warn('请选择时间后，在导出数据!!')
  }
  return(
    <div className='machine-download-container' >
      <div className='machine-download-flex' >
        <div className='machine-download-text' >导出历史数据</div>
        <DatePicker.RangePicker onChange={ getSelectTime } />
      </div>
      <Button className='machine-download-button' onClick={ tipsExcle } >
          导出excel
      </Button>
    </div>
    )
}

function getSelectTime(dates){
  dispatch('MACHINE_GET_DWONLOAD_EXCEL_TIME', dates)
}

// function downloadData(e){
//   console.log(e)
// }
