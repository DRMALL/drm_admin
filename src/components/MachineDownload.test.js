

import React from 'react'
import renderer from 'react-test-renderer'
import MachineDownload from './MachineDownload'

it('MachineDownload test', () => {
  const tree = renderer.create(<MachineDownload />)
  expect(tree).toMatchSnapshot()
})
