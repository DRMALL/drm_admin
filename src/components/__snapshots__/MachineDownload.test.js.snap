// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MachineDownload test 1`] = `
<div
  className="machine-download-container"
>
  <div
    className="machine-download-flex"
  >
    <div
      className="machine-download-text"
    >
      导出历史数据
    </div>
    <span
      className="ant-calendar-picker"
      style={Object {}}
    >
      <span
        className="ant-calendar-picker-input ant-input"
        disabled={undefined}
        onBlur={undefined}
        onClick={[Function]}
        onFocus={undefined}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        onTouchStart={[Function]}
      >
        <input
          className="ant-calendar-range-picker-input"
          disabled={undefined}
          placeholder="开始日期"
          readOnly={true}
          value=""
        />
        <span
          className="ant-calendar-range-picker-separator"
        >
           ~ 
        </span>
        <input
          className="ant-calendar-range-picker-input"
          disabled={undefined}
          placeholder="结束日期"
          readOnly={true}
          value=""
        />
        <span
          className="ant-calendar-picker-icon"
        />
      </span>
    </span>
  </div>
  <button
    className="ant-btn machine-download-button"
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span>
      导出excel
    </span>
  </button>
</div>
`;
